go/src/lib.rs

```diff
@@ -1,9 +1,162 @@
-/**
+/*
  * Copyright GLIDE-for-Redis Project Contributors - SPDX Identifier: Apache-2.0
  */
-use std::ffi::c_void;
+#![deny(unsafe_op_in_unsafe_fn)]
 
+use glide_core::client::Client as GlideClient;
+use glide_core::connection_request;
+use glide_core::errors;
+use glide_core::errors::RequestErrorType;
+use glide_core::ConnectionRequest;
+use protobuf::Message;
+use std::{
+    ffi::{c_void, CString},
+    os::raw::c_char,
+};
+use tokio::runtime::Builder;
+use tokio::runtime::Runtime;
+
+/// Success callback that is called when a Redis command succeeds.
+///
+/// `index` is the index of the operation which is completed.
```

Comment:
@acarbonetto's suggestion:
`index_ptr` is a baton-pass back to the caller language to uniquely identify the promise.